lib/reboot_cmds.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env python3
+
+"""
+Some uaclient operations cannot be fully completed by running a single
+command. For example, when upgrading uaclient from trusty to xenial,
+we may have a livepatch change in the contract, allowing livepatch to be
+enabled on xenial. However, during the upgrade we cannot install livepatch on
+the system, only after a reboot.
+
+To allow uaclient to postpone commands that need to be executed in a system boot,
+we are using this script, which will basically try to reprocess contract deltas
+on boot time.
+"""
+import logging
+import os
+import sys
+
+from uaclient import config
+
+from uaclient.util import subp, ProcessExecutionError
+from uaclient.cli import setup_logging, assert_lock_file
+
+
+BASE_PATH = "/var/lib/ubuntu-advantage/"
+REBOOT_CMD_MARKER_FILE = BASE_PATH + "marker-reboot-cmds-required"
+
+
+def run_command(cmd):
+    try:
+        out, _ = subp(cmd.split())
```

Comment:
Probably want to pass capture=True here to log the command and output to /var/log/ubuntu-advantage.log